sdk/storage/Azure.Storage.DataMovement.Files.Shares/src/DataMovementSharesExtensions.cs

```diff
@@ -352,6 +367,22 @@ internal static partial class DataMovementSharesExtensions
                 lastModifiedTime: shareItem?.Properties?.LastModified,
                 properties: properties);
         }
+
+        private static string[] ConvertContentPropertyObjectToStringArray(string contentPropertyName, object contentPropertyValue)
```

Comment:
Didn't we already have this function somewhere?

Reply:
You're thinking of the other one where it does it for just `string` not `string[]`

Reply:
see this I just made
https://github.com/Azure/azure-sdk-for-net/pull/45398/files#r1712077627

So I had to make this change cause the length wasn't being stored correctly, therefore not correctly seeing that there is a length in fact for downloading the file.